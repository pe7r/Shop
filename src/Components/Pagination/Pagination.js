import React, { Component } from 'react'
import './Pagination.scss'
import { leftArrow, rightArrow } from '../Icons/Icons'

export default class Pagination extends Component {
    render() {
        const { actualPage, totalCount, changingPage } = this.props;

        const pageCount = Math.ceil(totalCount/12);

        let disabledBack;
        if (actualPage === 1) {
            disabledBack = 'disabled'
        } 

        let disabledForward;
        if (actualPage === pageCount) {
            disabledForward = 'disabled'
        }
        return (
            
            <div className='pagination__container'>
                <button onClick={() => changingPage('prev')}
                        disabled={disabledBack}
                        name="buttonPrev"
                        className="pagination__button button-left">
                    {leftArrow}
                </button>
                <div className="pagination__info">
                    PAGE {actualPage} OF {pageCount}
                </div>
                <button onClick={() => changingPage('next')}
                        name="buttonNext"
                        disabled={disabledForward}
                        className="pagination__button button-right">
                    {rightArrow}
                </button>
            </div>
        )
    }
}
