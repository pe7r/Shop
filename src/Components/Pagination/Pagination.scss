.pagination__container {
    display: flex;
    
}