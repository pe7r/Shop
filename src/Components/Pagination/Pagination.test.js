import React from 'react';
import Pagination from './Pagination';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

describe('Pagination HTML renders correctly', () => {
    it('Pagionation with correct props', () => {
        const output = shallow(
            <Pagination actualPage={1} totalCount={879} changingPage={() => {}}/>
        )
        expect(shallowToJson(output)).toMatchSnapshot()
    })
})