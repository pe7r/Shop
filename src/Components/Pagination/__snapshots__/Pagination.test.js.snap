// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination HTML renders correctly Pagionation check with correct props 1`] = `
<div
  className="pagination__container"
>
  <button
    className="pagination__button button-left"
    disabled="disabled"
    name="buttonPrev"
    onClick={[Function]}
  >
    <svg
      fill="none"
      height="12"
      viewBox="0 0 21 12"
      width="21"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0.469669 5.46967C0.176777 5.76256 0.176777 6.23744 0.469669 6.53033L5.24264 11.3033C5.53553 11.5962 6.01041 11.5962 6.3033 11.3033C6.59619 11.0104 6.59619 10.5355 6.3033 10.2426L2.06066 6L6.3033 1.75736C6.59619 1.46447 6.59619 0.989594 6.3033 0.6967C6.01041 0.403807 5.53553 0.403807 5.24264 0.696701L0.469669 5.46967ZM21 5.25L1 5.25L1 6.75L21 6.75L21 5.25Z"
        fill="#056551"
      />
    </svg>
  </button>
  <div
    className="pagination__info"
  >
    PAGE 
    1
     OF 
    74
  </div>
  <button
    className="pagination__button button-right"
    name="buttonNext"
    onClick={[Function]}
  >
    <svg
      fill="none"
      height="12"
      viewBox="0 0 21 12"
      width="21"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M20.5303 6.53033C20.8232 6.23744 20.8232 5.76256 20.5303 5.46967L15.7574 0.696699C15.4645 0.403806 14.9896 0.403806 14.6967 0.696699C14.4038 0.989593 14.4038 1.46447 14.6967 1.75736L18.9393 6L14.6967 10.2426C14.4038 10.5355 14.4038 11.0104 14.6967 11.3033C14.9896 11.5962 15.4645 11.5962 15.7574 11.3033L20.5303 6.53033ZM0 6.75L20 6.75V5.25L0 5.25L0 6.75Z"
        fill="#056551"
      />
    </svg>
  </button>
</div>
`;
